doc/radosgw/troubleshooting.rst

```diff
@@ -175,3 +175,19 @@ Also, check to ensure that the default site is disabled. ::
   
   
   
+Numerous objects in default.rgw.meta pool
+=========================================
+
+Clusters created prior to *jewel* have a metadata archival feature enabled by default, using the ``default.rgw.meta`` pool.
+This archive keeps all old versions of user and bucket metadata, resulting in large numbers of objects in the ``default.rgw.meta`` pool.
+
+Users who want to disable this feature going forward should set the ``metadata_heap`` field to empty string ``""``::
+
+  $ radosgw-admin zone get --rgw-zone=default > zone.json
+  [edit zone.json, setting "metadata_heap": ""]
+  $ radosgw-admin zone set --rgw-zone=default --infile=zone.json
+  $ radosgw-admin period update --commit
+
+Next, users should confirm that ``default.rgw.meta`` is not used for any other purposes (e.g. ``user_keys_pool``, etc. in the default zone).
+
+Finally, users may safely delete the ``default.rgw.meta`` pool.
```

Comment:
thanks @dvanders! this is good, but i worry that a couple aspects could be confusing:

* there was a reshard bug in luminous/mimic that also left numerous stale bucket instance objects in this same default.rgw.meta pool - @theanalyst added a troubleshooting section to the resharding doc as part of https://github.com/ceph/ceph/pull/26564. we should probably combine both of these together here, as it's two possible causes of a similar issue

* in luminous, we consolidated a bunch of zone pools (domain_root, user_keys_pool, user_email_pool, user_swift_pool, user_uid_pool) into namespaces of the `default.rgw.meta` pool. you hinted at the possibility of the metadata_heap pool being shared with user_keys_pool, but it may not be clear that the namespace syntax `default.rgw.meta:root` implies sharing of the `default.rgw.meta` pool, rather than distinct pool names. i think we should spell that out clearly here, because deleting any of those could be catastrophic - and `rados ls -p default.rgw.meta` may look otherwise empty, since it doesn't list objects in namespaces. there's a description of pool namespaces in http://docs.ceph.com/docs/master/radosgw/pools/#pool-namespaces that can you link to for more info

Reply:
I've spelled out the namespaces concern -- what do you think about it now?

For the #26564 confusion, should I wait for them to move their doc in doc/radosgw/troubleshooting.rst then rebase?